src/com/gmail/luchyk/viktoriia/hw9/ArrayValueCalculator.java

```diff
@@ -0,0 +1,55 @@
+package com.gmail.luchyk.viktoriia.hw9;
+
+public class ArrayValueCalculator {
+    public static final int SIZE = 4;
+
+    public static int doCalc(String[][] data) throws ArraySizeException, ArrayDataException {
+        verify(data);
+        return sum(data);
+    }
+
+    public static int doCalculation(String[][] data) {
+        int result = 0;
+        try {
+            verify(data);
+        } catch (ArraySizeException e) {
+            System.out.printf("Incorrect size of the array. Should be %dx%d.\n", SIZE, SIZE);
+            return result;
+        }
+
+        try {
+            result = sum(data);
+        } catch (ArrayDataException e) {
+            System.out.println("Incorrect value in the array. Only numbers should be used.");
+            System.out.println(e);
+            return result;
+        }
+
+        return result;
+    }
+
+    private static void verify(String[][] data) throws ArraySizeException {
+        if (data.length != SIZE) throw new ArraySizeException();
+        else {
+            for (String[] row : data) {
+                if (row.length != SIZE) throw new ArraySizeException();
```

Comment:
In Java, we have a symmetrical array which means that if one row is 12  in size then all other rows will have a size of 12. Make no sense to iterate around all of them